src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -447,9 +472,28 @@ public bool Equals(Symbol? member1, Symbol? member2)
 
             // NB: up to, and including, this check, we have not actually forced the (type) parameters
             // to be expanded - we're only using the counts.
-            if (_considerArity && (member1.GetMemberArity() != member2.GetMemberArity()))
+            if (_considerArity)
             {
-                return false;
+                if (member1.GetMemberArity() != member2.GetMemberArity())
+                {
+                    return false;
+                }
+
+                if (member1.GetMemberArity() > 0 && (_considerParameterNames || _considerAttributes))
+                {
+                    ImmutableArray<TypeParameterSymbol> typeParams1 = member1.GetMemberTypeParameters();
+                    ImmutableArray<TypeParameterSymbol> typeParams2 = member2.GetMemberTypeParameters();
+
+                    if (_considerParameterNames && typeParams1.Length > 0 && !haveSameTypeParameterNames(typeParams1, typeParams2))
```

Comment:
>typeParams1.Length > 0 [](http://example.com/codeflow?start=51&length=22)

Can this be false?

Reply:
The thread is marked as resolved, but I do not see any response to the question. It was about ```typeParams1.Length > 0``` condition, BTW.